src/Calender.cpp

```diff
@@ -0,0 +1,93 @@
+
+#include "Calender.hpp"
+#include "utility"
+
+#include <algorithm>
+#include <stdexcept>
+#include <regex>
+
+auto Calender::findSchedule(User::Group group) const {
+    return schedules_.equal_range(group);
+}
+
+auto Calender::findLesson(const std::pair<Calender::Schedules::const_iterator, 
+                                          Calender::Schedules::const_iterator>& range,
+                          const std::string& date,
+                          const std::string& time) const
+{
+    auto prediction = [&date, &time](const auto& element)
+    {
+       return !element.second.getDate().compare(date) && !element.second.getTime().compare(time);
+    };
+    return std::find_if(range.first, range.second, prediction);
+}
+
+void Calender::validateDate(const std::string& date)
+{
+    std::regex dataMatch("[0-9]{2}[-|.|/][0-9]{2}[-|.|/][0-9]{4}|"
+                         "[0-9]{4}[-|.|/][0-9]{2}[-|.|/][0-9]{2}|");
+
+    if(!std::regex_match(date, dataMatch)) {
+        throw std::invalid_argument("invalid date format.\n"
+                                    "Valid formats are \"dd-mm-yyyy\" or \"yyyy-mm-dd\"\n"
+                                    "valid separetor \".\" \"-\" \"\\\"");
+    }
+}
+
+void Calender::validateTime(const std::string& time)
+{
+    std::regex timeMatch("[0-9]{2}[:][0-9]{2}");
```

Comment:
a little bit different regex should be used. This one will pass 99:99 which is not a proper time.
The same for date.

Reply:
Full regex implemented